### ISSUE TITLE
Main menu: double-clicking an empty content list causes a Lua runtime error

### ISSUE THREAD

**Issue by hlqkj** [minetest/minetest#14936]
### Minetest version

```true
Using LuaJIT 2.1.1706185428
Built by Clang 17.0
Running on Windows/10.0.22621 x86_64
BUILD_TYPE=Release
RUN_IN_PLACE=1
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="."
STATIC_LOCALEDIR="locale"
```

### Irrlicht device

SDL

### Operating system and version

Microsoft Windows 11 Pro, Version 23H2, OS Build 22631.3958, Codename is 'Copper, Sun Valley 3' methinks.

### CPU model

Intel Core i5-8250U @1.60GHz, Code Name 'Kaby Lake-R', Family 6, Model E, Stepping A, Ext. Family 6, Ext. Model 8E, Silicon Rev. Y0

### GPU model

NVIDIA GeForce MX150 (GP108), Rev. A1, Die size: 71mm², 1800M transistors, BIOS version 86.08.0E.00.7D, Driver version 31.0.15.4584 (NVIDIA 545.84) DCH / Win11 x64

### Active renderer

OpenGL 4.6.0

### Summary

> Describe your problem here.

It's not actually a problem of mine, but I thought to let you know, that if you double-click the "Installed Packages" list in the "Content" tab of the main menu (or hit enter on it), said list being empty, will cause the engine to say sad things.

I was not sure in which of the very relevant fields above to paste this, so I'll put it here:

```
Runtime error from mod '??' in callback handleMainMenuButtons(): ...-5.9.0-dev-win64\bin\..\builtin\mainmenu\tab_content.lua:198: attempt to index local 'pkg' (a nil value)
stack traceback:
	...-5.9.0-dev-win64\bin\..\builtin\mainmenu\tab_content.lua:198: in function 'handle_doubleclick'
	...-5.9.0-dev-win64\bin\..\builtin\mainmenu\tab_content.lua:219: in function 'handle_buttons'
	...ktop\minetest-5.9.0-dev-win64\bin\..\builtin\fstk\ui.lua:151: in function 'handle_buttons'
	...ktop\minetest-5.9.0-dev-win64\bin\..\builtin\fstk\ui.lua:195: in function <...ktop\minetest-5.9.0-dev-win64\bin\..\builtin\fstk\ui.lua:181>
```

### Steps to reproduce

> Explain how the problem has happened, providing a minimal test (e.g. a minimized code snippet) where possible.

I was just hanging around the main menu while drinking some coffee when, all of a sudden, the unthinkable happened.

A useful code snippet would be this one (offending line is 198):

https://github.com/minetest/minetest/blob/8ef2c42150123c4a626925f554dd74700c8b6dfe/builtin/mainmenu/tab_content.lua#L197-L210

**Comment by appgurueu** [comment]
This issue unfortunately goes deeper.

`pkg` is `nil` here because the double-click event carries an out of bounds row `1` despite the table being empty. This ultimately stems from parsing code in `guiFormspecMenu.cpp` which does a `split(parts[3], ',')`; split produces a vector containing just the empty string if given an empty string, as opposed to an empty vector, so tables / lists / dropdowns *think* they have at least one row when in reality they have zero (that's why you can "select") the first row.

In other words, this is a pretty clear engine bug. We should not be emitting `CHG` or `DCL` events for a nonexistant row. Instead, we should probably be emitting an `INV` ("no row selected") event, if anything.

This issue could be narrowly patched by adding a `nil` check here, but ideally we should fix the underlying engine bug.

**Comment by hlqkj** [comment]
Thanks for looking into this @appgurueu: I was very close to open a PR to add that `nil` check, but I could not fully understand why that was happening, so I stepped back and just reported it.
